pkgs/applications/editors/jetbrains/update_ides.py

```diff
@@ -101,6 +105,27 @@ def update_products(products):
     json.dump(versions, versions_file, indent=2)
     versions_file.write("\n")
 
+if len(toVersions) == 0:
+    # No Updates found
+    sys.exit(0)
+
+if len(toVersions) == 1:
+    commitMessage = ""
+else:
```

Comment:
how does `toVersions` ever reach a number >1?  the only time it's written to is:
```
                toVersions[name] = new_version
```
in line 80.

What am I missing?

Reply:
This might happen, if there are updates for more than one Product. Right at the moment there is an update for `rider` and `phpstorm`, so `len(toVersions)` is 2.

In this case, `toVersions` looks similar to this:
![image](https://github.com/NixOS/nixpkgs/assets/1544760/a006ed7c-ebc1-43ae-bf68-f06b27c432c3)

This piece of code is to write different commit messages. In case there is exactly one product update, we'll write that as "subject" (e.g. `jetbrains.rider: 2023.1.4 -> 2023.2`). If there are more Products updated, we'll write an additional summary:
```
jetbrains: 2023.1.4 -> 2023.2 // jetbrains: <min version of updated products> -> <max version of updated products>
    
jetbrains.phpstorm: 2023.1.4 -> 2023.2
jetbrains.rider: 2023.1.4 -> 2023.2
```

This could be simplified if we drop the summary. Then we would not need that min/max calculation, and it could be just a list (string) of updated products.

Reply:
Ah I missed that it does multiple versions at once, yeah you are right.